### PR TITLE
Default new project to the ios-signing-cert development team

### LINKED ISSUE #90020: Default new iOS Xcode projects to the Developer Team saved in config "ios-signing-cert"
If `flutter config` `ios-signing-cert` is stored, use the Developer Team identifier when `flutter create`ing a new iOS app.  This will allow `ios/Runner.xcworkspace` to be runnable on a real device without needing to change Xcode build settings.

See `Team` instructions in https://flutter.dev/docs/deployment/ios#review-xcode-project-settings.

### PR DESCRIPTION
**Pull request by jmagman**
Set `DEVELOPMENT_TEAM` Xcode build setting if the Developer Team is already saved in `flutter config`.

Fixes https://github.com/flutter/flutter/issues/90020

### CODE REVIEW

File: packages/flutter_tools/lib/src/ios/code_signing.dart [hunk 1]
```diff
@@ -193,28 +233,24 @@ Future<Map<String, String>?> getCodeSigningIdentityDevelopmentTeam({
...
   Logger logger,
   Config config,
   Terminal terminal,
+  bool shouldExitOnNoCerts,
 ) async {
   // The user has no valid code signing identities.
   if (validCodeSigningIdentities.isEmpty) {
-    logger.printError(noCertificatesInstruction, emphasis: true);
-    throwToolExit('No development certificates available to code sign app for device deployment');
+    if (shouldExitOnNoCerts) {
```
Comment (jmagman) [inline packages/flutter_tools/lib/src/ios/code_signing.dart:248]: This is a little clunky, but the old code was called when a code signing cert was actually necessary on build, and the new code wants to non-fatally check if there's a saved cert, and if there isn't just skip using any in the create template.

File: packages/flutter_tools/lib/src/ios/code_signing.dart [hunk 2]
```diff
@@ -116,6 +118,53 @@ Future<Map<String, String>?> getCodeSigningIdentityDevelopmentTeam({
...
+  if (developmentTeam == null) {
+    return null;
+  }
+
+  return <String, String>{
+    'DEVELOPMENT_TEAM': developmentTeam,
+  };
+}
+
+Future<String?> getCodeSigningIdentityDevelopmentTeam({
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/ios/code_signing.dart:139]: Do we need this wrapper, or can we make `_getCodeSigningIdentityDevelopmentTeam` public?

File: packages/flutter_tools/lib/src/ios/code_signing.dart [hunk 3]
```diff
@@ -116,6 +118,53 @@ Future<Map<String, String>?> getCodeSigningIdentityDevelopmentTeam({
...
+/// Set [shouldExitOnNoCerts] to show instructions for how to add a cert when none are found, then [toolExit].
+Future<String?> _getCodeSigningIdentityDevelopmentTeam({
+  required ProcessManager processManager,
+  required Platform platform,
+  required Logger logger,
+  required Config config,
+  required Terminal terminal,
+  bool shouldExitOnNoCerts = false,
+}) async {
+  if (!platform.isMacOS) {
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/ios/code_signing.dart:164]: should this be an exception?

File: packages/flutter_tools/lib/src/ios/code_signing.dart (hunk 2 above)
Comment (jmagman) [reply packages/flutter_tools/lib/src/ios/code_signing.dart:139]: I was trying to hide the implementation detail of `shouldExitOnNoCerts` from the public APIs it's true for `getCodeSigningIdentityDevelopmentTeamBuildSetting` and false for `getCodeSigningIdentityDevelopmentTeam`.

File: packages/flutter_tools/lib/src/ios/code_signing.dart (hunk 1 above)
Comment (christopherfujino) [reply packages/flutter_tools/lib/src/ios/code_signing.dart:258]: can you add a comment giving a hint about the `shouldExitOnNoCerts == false` behavior?

File: packages/flutter_tools/test/commands.shard/permeable/build_apk_test.dart [hunk 4]
```diff
@@ -129,7 +129,7 @@ void main() {
 
     group('AndroidSdk', () {
       testUsingContext('throws throwsToolExit if AndroidSdk is null', () async {
-        final String projectPath = await createProject(tempDir, arguments: <String>['--no-pub', '--template=app']);
+        final String projectPath = await createProject(tempDir, arguments: <String>['--no-pub', '--template=app', '--platform=android']);
```
Comment (christopherfujino) [inline packages/flutter_tools/test/commands.shard/permeable/build_apk_test.dart:132]: why did you need to update these tests?

File: packages/flutter_tools/lib/src/ios/code_signing.dart (hunk 2 above)
Comment (christopherfujino) [reply packages/flutter_tools/lib/src/ios/code_signing.dart:139]: makes sense

File: packages/flutter_tools/lib/src/commands/create.dart [hunk 5]
```diff
@@ -231,6 +232,17 @@ class CreateCommand extends CreateBase {
     }
 
     final String dartSdk = globals.cache.dartSdkBuild;
+    final bool includeIos = featureFlags.isIOSEnabled && platforms.contains('ios');
+    String developmentTeam;
```
Comment (cyanglaz) [inline packages/flutter_tools/lib/src/commands/create.dart:236]: How about init it with '', so later, we only need to check `if (iosDevelopmentTeam.isNotEmpty)`, it would also fit better with NNBD, and make the migration easier. 

And nits: maybe iosDevelopmentTeam would be a better name as it was only used for iOS.

File: packages/flutter_tools/lib/src/commands/create_base.dart [hunk 6]
```diff
@@ -375,6 +376,8 @@ abstract class CreateBase extends FlutterCommand {
       'withPluginHook': withPluginHook,
       'androidLanguage': androidLanguage,
       'iosLanguage': iosLanguage,
+      'hasIosDevelopmentTeam': iosDevelopmentTeam != null && iosDevelopmentTeam.isNotEmpty,
```
Comment (cyanglaz) [inline packages/flutter_tools/lib/src/commands/create_base.dart:379]: Can just be `'hasIosDevelopmentTeam': iosDevelopmentTeam.isNotEmpty` if taking my suggestion above.

File: packages/flutter_tools/lib/src/commands/create_base.dart [hunk 7]
```diff
@@ -375,6 +376,8 @@ abstract class CreateBase extends FlutterCommand {
...
+      'iosDevelopmentTeam': iosDevelopmentTeam ?? '',
```
Comment (cyanglaz) [inline packages/flutter_tools/lib/src/commands/create_base.dart:380]: Can just be `'iosDevelopmentTeam': iosDevelopmentTeam,` if taking my suggestion above.

File: packages/flutter_tools/lib/src/ios/code_signing.dart [hunk 8]
```diff
@@ -116,6 +118,53 @@ Future<Map<String, String>?> getCodeSigningIdentityDevelopmentTeam({
...
+    terminal: terminal,
+    shouldExitOnNoCerts: true,
+  );
+
+  if (developmentTeam == null) {
+    return null;
+  }
+
+  return <String, String>{
+    'DEVELOPMENT_TEAM': developmentTeam,
```
Comment (cyanglaz) [inline packages/flutter_tools/lib/src/ios/code_signing.dart:135]: nits: Have 'DEVELOPMENT_TEAM' assigned to a constant and use the constant instead.

File: packages/flutter_tools/templates/module/ios/host_app_ephemeral/Runner.xcodeproj.tmpl/project.pbxproj.tmpl [hunk 9]
```diff
@@ -437,6 +443,9 @@
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CURRENT_PROJECT_VERSION = "$(FLUTTER_BUILD_NUMBER)";
+{{#hasIosDevelopmentTeam}}
+				DEVELOPMENT_TEAM = {{iosDevelopmentTeam}};
```
Comment (cyanglaz) [inline packages/flutter_tools/templates/module/ios/host_app_ephemeral/Runner.xcodeproj.tmpl/project.pbxproj.tmpl:447]: Just curious, what happens for `DEVELOPMENT_TEAM = "";`, would it be cleaner that DEVELOPMENT_TEAM is set to an empty string if the developer didn't set it?

**Review by cyanglaz — APPROVED** [review]
LGTM % some nits

File: packages/flutter_tools/test/commands.shard/permeable/build_apk_test.dart (hunk 4 above)
Comment (jmagman) [reply packages/flutter_tools/test/commands.shard/permeable/build_apk_test.dart:132]: So I didn't have to mock out the iOS signing mechanism in an Android-only test.

Comment (christopherfujino) [reply packages/flutter_tools/test/commands.shard/permeable/build_apk_test.dart:132]: :+1:

File: packages/flutter_tools/lib/src/commands/create.dart (hunk 5 above)
Comment (christopherfujino) [reply packages/flutter_tools/lib/src/commands/create.dart:236]: I'd prefer if we left it as uninitialized, so we can differentiate between explicitly set as an empty string and never set.

Comment (cyanglaz) [reply packages/flutter_tools/lib/src/commands/create.dart:236]: Is there a specific meaning of development team explicitly set as empty string? @jmagman Might be able to answer that. I was looking at the line below `'iosDevelopmentTeam': iosDevelopmentTeam ?? '',` thinking that they should be the same.

**Review by christopherfujino — APPROVED** [review]
LGTM
